site/source/docs/api_reference/wasm_workers.rst

```diff
@@ -0,0 +1,355 @@
+``emscripten_dispatch_to_thread()``, etc. from ``emscripten/threading.h`` to perform proxied calls.
+
+Wasm Workers do not provide this functionality. If needed, such messaging should be implemented manually
+by users via regular multithreaded synchronized programming techniques (mutexes, futexes, semaphores, etc.)
+
+Pthreads synchronize wallclock times
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Another portability aiding emulation feature that Pthreads provide is that the time values returned by
+``emscripten_get_now()`` are synchronized to a common time base across all threads.
+
+Wasm Workers omit this concept, and it is recommended to use the function ``emscripten_performance_now()``
+for high performance timing in a Wasm Worker, and avoid comparing resulting values across Workers, or
+manually synchronize them.
+
+Input events API backproxies only to pthreads
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The multithreaded input API provided in ``emscripten/html5.h`` only works with the pthread API. When
+calling any of the functions ``emscripten_set_*_callback_on_thread()``, one can choose the target
+pthread to be the recipient of the received events.
+
+With Wasm Workers, if desired, "backproxying" events from the main browser thread to a Wasm Worker
+should be implemented manually e.g. by using the ``emscripten_wasm_worker_post_function_*()`` API family.
+
+However note that backproxying input events has a drawback that it prevents security sensitive operations,
+like fullscreen requests, pointer locking and audio playback resuming, since handling the input event
+is detached from the event callback context executing the initial operation.
+
+Pthread vs emscripten_lock implementation differences
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The mutex implementation from ``pthread_mutex_*`` has a few different creation options, one being a
+"recursive" mutex.
+
+The lock implemented by ``emscripten_lock_*`` API is not recursive (and does not provide an option).
+
+Pthreads also offer a programming guard against a programming error that one thread would not release
+a lock that is owned by another thread. ``emscripten_lock_*`` API does not track lock ownership.
+
+Memory requirements
+^^^^^^^^^^^^^^^^^^^
+
+Pthreads have a fixed dependency to dynamic memory allocation, and perform calls to ``malloc`` and ``free``
+to allocate thread specific data, stacks and TLS slots.
+
+With the exception of the helper function ``emscripten_malloc_wasm_worker()``, Wasm Workers are not dependent
+on a dynamic memory allocator. Memory allocation needs are met by the caller at Worker creation time, and
+can be statically placed if desired.
+
+Generated code size
+^^^^^^^^^^^^^^^^^^^
+
+The disk size overhead from pthreads is on the order of a few hundred KBs. Wasm Workers runtime on the other
+hand is optimized for tiny deployments, just a few hundred bytes on disk.
```

Comment:
NOrmally I think of code size in terms of wire bytes. Is that different from the "on disk" code size discussed here?

Reply:
I suppose the wire size and disk size are generally gzipped/brotlied vs uncompressed. The 'on disk' here just colloquially means to the uncompressed output size of the file.